runtime/commonMain/src/kotlinx/serialization/Polymorphic.kt

```diff
@@ -75,5 +74,12 @@ public val PolymorphicClassDescriptor = SerialDescriptor("kotlinx.serialization.
  * @see SerializersModuleBuilder.polymorphic
  */
 public class PolymorphicSerializer<T : Any>(override val baseClass: KClass<T>) : AbstractPolymorphicSerializer<T>() {
-    public override val descriptor: SerialDescriptor = PolymorphicClassDescriptor
+    public override val descriptor: SerialDescriptor =
+        SerialDescriptor("kotlinx.serialization.Polymorphic", PolymorphicKind.OPEN) {
+            element("type", String.serializer().descriptor)
+            element(
+                "value",
+                SerialDescriptor("kotlinx.serialization.Polymorphic.${baseClass.simpleName}", UnionKind.CONTEXTUAL)
```

Comment:
Added class simple name for the SD FQN for the sake of observability and debuggability.
It doesn't really matter, just makes things more straightforward

Reply:
Maybe
```suggestion
                SerialDescriptor("kotlinx.serialization.Polymorphic<${baseClass.simpleName}>", UnionKind.CONTEXTUAL)
```
to be less cryptic?